src/h3lib/lib/h3Index.c

```diff
@@ -217,38 +223,38 @@ H3Index makeDirectChild(H3Index h, int cellNumber) {
 /**
  * cellToChildren takes the given hexagon id and generates all of the children
  * at the specified resolution storing them into the provided memory pointer.
- * It's assumed that maxCellToChildrenSize was used to determine the allocation.
+ * It's assumed that cellToChildrenSize was used to determine the allocation.
  *
  * @param h H3Index to find the children of
  * @param childRes int the child level to produce
  * @param children H3Index* the memory to store the resulting addresses in
  */
 void H3_EXPORT(cellToChildren)(H3Index h, int childRes, H3Index* children) {
-    int parentRes = H3_GET_RESOLUTION(h);
-    if (!_isValidChildRes(parentRes, childRes)) {
-        return;
-    } else if (parentRes == childRes) {
-        *children = h;
-        return;
-    }
-    int bufferSize = H3_EXPORT(maxCellToChildrenSize)(h, childRes);
-    int bufferChildStep = (bufferSize / 7);
-    int isAPentagon = H3_EXPORT(isPentagon)(h);
-    for (int i = 0; i < 7; i++) {
-        if (isAPentagon && i == K_AXES_DIGIT) {
-            H3Index* nextChild = children + bufferChildStep;
-            while (children < nextChild) {
-                *children = H3_NULL;
-                children++;
-            }
-        } else {
-            H3_EXPORT(cellToChildren)
-            (makeDirectChild(h, i), childRes, children);
-            children += bufferChildStep;
-        }
+    IterCellsChildren iter = iterInitParent(h, childRes);
+
+    for (int64_t i = 0; iter.h; i++, iterStepChild(&iter)) {
+        children[i] = iter.h;
     }
 }
 
+/**
+ * Zero out index digits from start to end, inclusive.
+ * No-op if start > end.
+ */
+H3Index _zeroIndexDigits(H3Index h, int start, int end) {
+    if (start > end) return h;
+
+    uint64_t m = 0;
```

Comment:
nit: Make `m` of type `H3Index`?

Reply:
I'm not sure that would particularly help clarity. I suggest H3Index to indicate in the code that these are exactly the same types (same width, and so on). I don't think having an additional type makes a lot of sense since C does not have the kind of semantic typing that would make that really useful.

Reply:
I change to `H3Index`, but that feels less clear to me as I don't actually think of these as having the same *type*, even though they do have the same *size*. But I also don't feel that strongly about it. :)

Reply:
I'll just state that I agree with Isaac here. In my opinion the bitmasking manipulation *only* makes sense if the type is `H3Index`, not `int64_t` since `H3Index` indicates meaning towards particular bit fields in a way that `int64_t` does not.

Reply:
Fair enough. I'll move to use `H3Index`.